### ISSUE TITLE
Problem with TrainingHelper's parameter sequence length 

### ISSUE THREAD

**Issue by yangshao** [j-min/tf_tutorial_plus#3]
hi, I'm trying to implement a seq2seq model. when I use the actual sequence length as TrainingHelper's parameter, I get an error for the loss calculation:
InvalidArgumentError (see above for traceback): logits and labels must have the same first dimension, got logits shape [960,15038] and labels shape [1280]
But if I use all the same sequence length, the code works fine. what's the effects of using actual sequence length instead of using max sequence length?

**Comment by jihopark** [comment]
i've got a similar problem. i think the decoder is sometimes outputting logits of shorter length than the max seq length.

**Comment by yangshao** [comment]
@jihopark do you solve this problem?

**Comment by j-min** [comment]
Did you check arguments of `tf.contrib.seq2seq.dynamic_decode` as below?
`impute_finished = True`
`maximum_sequence_length=dec_sentence_length (int, maximum length of current batch)`

```
tf.contrib.seq2seq.dynamic_decode(
                    inference_decoder,
                    output_time_major=False,
                    impute_finished=True,
                    maximum_iterations=dec_sentence_length)
```

**Comment by yangshao** [comment]
yes，i checked it. besides， do you know howto do beam search in tf 1.1.0？
thanks
